chore: add k8s 1.24+ taints

frezbo opened the pull request:
Add new `node-role.kubernetes.io/control-plane` taints so that the
cloud controller tolerates them.

Signed-off-by: Noel Georgi <git@frezbo.dev>

Fixes: #430

kwohlfahrt commented:
It would only affect verions <= 1.19, which went EOL October 2021, so I would support just moving forward. However on EKS 1.19 is supported until August 2022.

frezbo commented:
@kwohlfahrt I have updated the PR and ended up using `nodeAffinity` since it's OR'ed. So this should still work for old k8s. I've also fixed up the helm values file, so it installs properly out of the box.

frezbo commented:
/assign

frezbo commented:
/assign @nckturner

mikescholl-sysdig commented:
hI @frezbo @nckturner any chance this is getting pushed forward? Currently running into this in a couple clusters where the `master` role no longer exists.

frezbo commented:
@mikescholl-sysdig I'm also waiting on the maintainers

nckturner commented:
Sorry for the delay.  

/triage accepted
/ok-to-test

frezbo commented:
```bash
 /home/prow/go/src/k8s.io/cloud-provider-aws
./hack/e2e/run.sh: line 149: /home/prow/go/src/k8s.io/cloud-provider-aws/_output/test/20220819181918/kops: No such file or directory 
```

seems like an unrelated failure? :thinking:

olemarkus commented:
/test pull-cloud-provider-aws-e2e

nckturner commented:
@frezbo you'll need to rebase, the above failure was fixed in master.

nckturner commented:
We should bump the chart version as well, shouldn't we?

frezbo commented:
> We should bump the chart version as well, shouldn't we?

Does 0.0.7 sound fine?

frezbo commented:
Updated the chart version to `0.0.7` and also fixed the e2e test failure.

frezbo commented:
@nckturner this should be good to go

frezbo commented:
closing in favor of https://github.com/kubernetes/cloud-provider-aws/pull/466

olemarkus commented:
Sorry I missed this one. I was sort of going through my mailbox latest to oldest and #466 appeared first.

mikescholl-sysdig commented:
@olemarkus FYI #466 is not an ideal solution since you will now break K8S versions older than 1.24 that do not have the control-plane taints. It is also impossible to clear that setting when its set in the values.yaml
